### ISSUE TITLE
Language picker in master doesn't apper to be working (but it does)

### ISSUE THREAD

**Issue by diramazioni** [mirumee/saleor#3328]
### What I'm trying to achieve
As you can see going to the [live demo](https://demo.getsaleor.com/pl/) you can't switch language from the language picker

However in my local instance while customizing i18n messages and running as in [the doc](https://docs.getsaleor.com/en/latest/customization/i18n.html)
`python manage.py makemessages -l en --extension=email,html,mjml,py,txt --ignore="templates/templated_email/compiled/*"`
and
`$ python manage.py compilemessages`
seems to have simply fixed the issue.

I think this it's a minor issue since the dev will end up running makemessage/ compilemessages anyway while customizing it like I did. 
Can you check if the issue is actually simply solved by simply running makemessage/ compilemessages commands?
I've noticed that the django.po messages in the master are out of sync anyway so the need rebuild

**Comment by NyanKiyoshi** [comment]
## Steps to reproduce
1. Go to https://demo.getsaleor.com/pl/;
1. Try to switch the language to French;
1. It should set it (you can check through the response headers), but redirects to `/pl/`;
1. Try to go to `https://demo.getsaleor.com/`, it should redirect to `https://demo.getsaleor.com/fr/`.

## Checklist
- [x] Does it happen on the master branch or only the demo branch?
- [ ] Does it happen with a given django version or another requirement?

## Notes
There is a test that is supposed to ensure this issue doesn't occur.
https://github.com/mirumee/saleor/blob/f9fdf5ed2f1e79f7e784076933aaab37b002baa0/tests/test_core.py#L213-L217

**Comment by NyanKiyoshi** [comment]
I manage to reproduce it on master. Here is how:

1. Switch from English to French;
1. Open the endpoint `/en/`;
1. Switch to Polish;
1. We should be redirected to `/en/` (to be able to switch languages, we have to return to `/`).

This is an issue that users can run into from mirumee/saleor#3154, because the user gets redirect to `/en/` instead of the set language.

**Comment by diramazioni** [comment]
Yes, this what it is!
[can this be helpful?](https://stackoverflow.com/questions/27680748/when-using-i18n-patterns-how-to-reverse-url-without-language-code)

**Comment by NyanKiyoshi** [comment]
The issue is not really there, I looked the issue is directly from Django. It looks like it's a wanted behavior from django to force the language to be the old one if the current language is not the set one (the one set in the cookies).

So, I don't know. It would require to change the logic from Django's `reverse` function (called from `set_language()`). Maybe it would be easier to open a ticket or something to know how we could prevent this behavior.

Or otherwise, it could be possible to match [the next URL against the i18n pattern](https://github.com/django/django/blob/f091ea35150d95fc6732bbf0c27b971dd445509a/django/views/i18n.py#L46) and drop it (e.g.: `/en/` => `/`) to force django to do a 302 to whatever language is set. But that would be hacky.

TL;DR: the faulty part is from `reverse()` of `translate_url()`. Looks like a wanted behavior.

**Comment by diramazioni** [comment]
Sorry my skills with django are pretty basics to follow up, but I've seen similar problems affecting other projects with django like mezzanine. I don't know how they end up solving it.

Btw can't we bypass django redirection entirely by setting it client side using Web Storage?

**Comment by NyanKiyoshi** [comment]
We would lose whatever I18N features django has on the language cookies then. That would not be a solution.

I could take a look at this issue around the end of December, if I can find a solution (or that least poke around for a solution) that would not be hacky, so we have something good and clean, that works for in every cases. Especially as this issue is minor (as of now at least). If we could fix mirumee/saleor#3154 in before (higher priority), that would be awesome though.
